DR2-1915 Send ingest started message.

MancunianSam opened the pull request:
When the event type is insert and the inserted item is an asset, send an
ingest started message.

techncl reviewed (commented):
Looks good! Just one comment as well as:

1. Do you think it's worth adding a test in `LambdaTest` for "handler" should "not send an ingest started message if the event name is not insert and the type is an asset"
2. Should we ensure that the `MODIFY` step comes after the `INSERT` stuff
3. Are the `messageType` changes [you made](https://github.com/nationalarchives/dr2-ingest/pull/278), going to conflict with these?
4. Could you update the README?

MancunianSam commented:
1. We do sometimes send a message if the type is MODIFY and it's an asset which is what the other tests are checking. I've added one in to check we don't send a message for REMOVE as that's all that's missing.
2. Why? They're sending different messages so even if we somehow got an insert and a modify at the same time, we'd send two messages as expected.
3. That branch was set to merge into this one which I did so the messageType which I forgot when I merged so these changes are here now :) 
4. I've added in the new test cases.

MancunianSam commented on scala/lambdas/utils/src/main/scala/uk/gov/nationalarchives/utils/ExternalUtils.scala:
```diff
@@ -179,6 +179,7 @@ object ExternalUtils {
   enum MessageStatus(val value: String):
     case IngestedPreservation extends MessageStatus("Asset has been ingested to the Preservation System.")
     case IngestedCCDisk extends MessageStatus("Asset has been written to custodial copy disk.")
+    case IngestStarted extends MessageStatus("Asset has started the ingest process")
```
I've added them in.

MancunianSam commented on scala/lambdas/ingest-files-change-handler/src/main/scala/uk/gov/nationalarchives/ingestfileschangehandler/Lambda.scala:
```diff
@@ -29,19 +30,16 @@ class Lambda extends LambdaRunner[DynamodbEvent, Unit, Config, Dependencies]:
 
     def sendMessageAndDelete(asset: AssetDynamoItem, children: List[FileDynamoItem]) = for {
-      _ <- sendOutputMessage(asset, IngestComplete)
+      _ <- sendOutputMessage(asset, IngestComplete, MessageStatus.IngestedCCDisk)
```
There's three possibilities but it's a fair point, I've created three methods, `sendIngestedPreservation`, `sendIngestedCCDisk` and `sendIngestStarted`

techncl commented:
> 2. Why? They're sending different messages so even if we somehow got an insert and a modify at the same time, we'd send two messages as expected.

Oh, because you're using `>>` so I assumed that the `INSERT` has to come before the `MODIFY`. Could these be done in parallel then or does it not matter?

MancunianSam commented:
Good point, it can be parallel. I've changed it so that each record starts its own fiber.

techncl commented:
Looks good!

techncl reviewed (approved):
👍
